lib/puma/io_buffer.rb

```diff
@@ -1,4 +1,19 @@
 # frozen_string_literal: true
 
-require 'puma/detect'
-require 'puma/puma_http11'
+module Puma
+  class IOBuffer < String
+    def initialize
+      if RUBY_VERSION < "2.2"
+        super
+      else
+        super(capacity: 4096)
+      end
+    end
+
+    def append(*args)
+      args.each { |a| concat(a) }
```

Comment:
FWIW, Ruby 2.4+ supports multi-args `String#concat`.

Reply:
Nice! We have to support 2.2+ for now.